### PR TITLE
rebased

### PR DESCRIPTION
**Pull request by WuTheFWasThat**

### CODE REVIEW

File: assets/js/data.coffee [hunk 1]
```diff
@@ -165,6 +168,8 @@ class Data
     allMarks = do @store.getAllMarks
 
     if mark of allMarks
+      if allMarks[mark] == row.id
```
Comment (vanceza) [inline assets/js/data.coffee]: Looks like it should be in master

File: assets/js/data.coffee [hunk 2]
```diff
@@ -267,26 +295,37 @@ class Data
...
 
-    @detachMarks row # Requires parent to be set correctly, so it's at the beginning
+    if @exactlyOneInstance row.id # If detaching the LAST instance
+      @detachMarks row # Requires parent to be set correctly, so it's at the beginning
 
     parent = do row.getParent
     children = @getSiblings row
-    i = @indexOf row
-    children.splice i, 1
+    ci = @indexOf row
```
Comment (vanceza) [inline assets/js/data.coffee]: Should be in master

Comment (vanceza) [inline assets/js/data.coffee]: Whoops, didn't notice parent -> parents migration was in this, nm

**Comment by WuTheFWasThat** [comment]
cool, looks good

**Comment by WuTheFWasThat** [comment]
i just noticed i messed up the author on 5180755, feel free to amend it

**Pull request closed without merging**